model/src/main/kotlin/licenses/ResolvedLicenseInfo.kt

```diff
@@ -73,6 +77,40 @@ data class ResolvedLicenseInfo(
             resolvedLicense.originalExpressions.map { it.expression }
         }.andOrNull()
 
+    /**
+     * Return the main license of a package (or project) as an [SpdxExpression], or null if there is no main license.
+     * The main license is the conjunction of the declared license of the package (or project) and the licenses detected
+     * in any of the configured [LicenseFilePatterns] matched against the root path of the package (or project).
+     */
+    fun mainLicense(): SpdxExpression? {
+        val matcher = PathLicenseMatcher(LicenseFilePatterns.getInstance())
+        val licensePaths = flatMap { resolvedLicense ->
+            resolvedLicense.locations.map { it.location.path }
+        }
+
+        val detectedLicenses = filterTo(mutableSetOf()) { resolvedLicense ->
+            resolvedLicense.locations.any {
+                val rootPath = (it.provenance as? RepositoryProvenance)?.vcsInfo?.path.orEmpty()
+
+                val applicableLicensePaths = matcher.getApplicableLicenseFilesForDirectories(
+                    licensePaths,
+                    listOf(rootPath)
+                )
+
+                val applicableLicenseFiles = applicableLicensePaths[rootPath].orEmpty()
+
+                it.location.path in applicableLicenseFiles
+            }
+        }
+
+        val declaredLicenses = filter(LicenseView.ONLY_DECLARED)
+        val mainLicenses = (detectedLicenses + declaredLicenses.licenses)
+            .flatMap { it.originalExpressions }
+            .mapTo(mutableSetOf()) { it.expression }
```

Comment:
For performance reasons, I propose to drop simply map this to a list so that `distinct()` is called only once, within `toExpression()`.